bundle/src/main/groovy/org/cid15/aem/groovy/console/replication/ReplicatedScriptListener.groovy

```diff
@@ -0,0 +1,71 @@
+package org.cid15.aem.groovy.console.replication
+
+import com.day.cq.commons.jcr.JcrConstants
+import com.google.common.base.Charsets
+import groovy.util.logging.Slf4j
+import org.apache.sling.api.resource.ResourceResolver
+import org.apache.sling.api.resource.ResourceResolverFactory
+import org.apache.sling.api.resource.observation.ResourceChange
+import org.apache.sling.api.resource.observation.ResourceChangeListener
+import org.cid15.aem.groovy.console.GroovyConsoleService
+import org.cid15.aem.groovy.console.api.context.ScriptContext
+import org.cid15.aem.groovy.console.api.impl.ResourceScriptContext
+import org.jetbrains.annotations.NotNull
+import org.osgi.service.component.annotations.Component
+import org.osgi.service.component.annotations.Reference
+
+import javax.jcr.Session
+
+import static com.google.common.base.Preconditions.checkNotNull
+
+// TODO: Use const references for properties
+@Component(property = [
+        "resource.paths=/conf/groovyconsole/replication",
+        "resource.change.types=ADDED"
+])
+@Slf4j("LOG")
+public class ReplicatedScriptListener implements ResourceChangeListener {
```

Comment:
Create an osgi property to enable / disable this feature

Reply:
why? I think we would only need to disable on author

Reply:
If anyone could POST a file in this directory, because they left this open in the Dispatcher, this would be a security risk that could not be fixed. I'd rather not be like Adobe where the only option to not expose all info is by blocking it in the dispatcher.

Defence by depth

Reply:
agreed, will add